### PR TITLE
Change default format of carrington_rotation_time return

### PR DESCRIPTION
**Pull request by dstansby**
Fixes https://github.com/sunpy/sunpy/issues/4818 - I wasn't sure how to change the scale too here as https://github.com/sunpy/sunpy/issues/4818#issuecomment-754057830 suggests, but happy to with some pointers.

### CODE REVIEW

File: changelog/4819.trivial.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Change the format of the time returned from `~sunpy.coordinates.sun.carrington_rotation_number`
```
Comment (ayshih) [inline changelog/4819.trivial.rst:1]:
```suggestion
Change the format of the time returned from :func:`~sunpy.coordinates.sun.carrington_rotation_number`
```

File: sunpy/coordinates/sun.py [hunk 2]
```diff
@@ -121,7 +121,9 @@ def refine(estimate):
     # Perform two iterations of the correction to achieve sub-second accuracy
     estimate = refine(estimate)
     estimate = refine(estimate)
-    return Time(estimate, scale='tt', format='jd')
+    t = Time(estimate, scale='tt', format='jd')
```
Comment (ayshih) [inline sunpy/coordinates/sun.py:124]:
```suggestion
    t = Time(estimate, scale='tt', format='jd').utc
```

File: sunpy/coordinates/tests/test_sun.py [hunk 3]
```diff
@@ -479,3 +479,9 @@ def test_carrington_rotation_roundtrip():
     dt = t - t_roundtrip
     # Stated precision in the docstring is 0.11 seconds
     assert_quantity_allclose(dt.to(u.s), 0*u.s, atol=0.11*u.s)
+
+
+def test_carrington_rotation_str():
+    # Check that by default a human parseable string is returned
+    t = sun.carrington_rotation_time(2210)
+    assert str(t) == '2018-10-26 20:49:25.321'
```
Comment (ayshih) [inline sunpy/coordinates/tests/test_sun.py:487]:
```suggestion
    assert str(t) == '2018-10-26 20:48:16.137'
```

**Review by ayshih — CHANGES_REQUESTED** [review]

**Comment by Cadair** [comment]
Is the scale change here actually more of a change? This will now display different numbers for someone who prints it in `iso` format than before?

**Comment by nabobalis** [comment]
Is it worth backporting to 2.0?

**Comment by dstansby** [comment]
I'd say no, since it's not a trivial change.

**Review by nabobalis — APPROVED** [review]

File: changelog/4819.trivial.rst [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Change the format of the time returned from :func:`~sunpy.coordinates.sun.carrington_rotation_number`
```
Comment (Cadair) [inline changelog/4819.trivial.rst:1]: This should mention the scale change as well. (Maybe not trivial?)

**Comment by dstansby** [comment]
I put a changelog in `breaking` just to be safe

**Review by Cadair — APPROVED** [review]

**Review by ayshih — APPROVED** [review]

**Pull request merged**